### PR TITLE
Fix read_attribute_for_serialization not seeing parent serializer methods

### LINKED ISSUE #1653: Inherited methods are hidden in serializer
#### Expected behavior vs actual behavior
When subclassing a serializer, the parent class's instance methods are not populated in the child class's instance method list, so `_serializer_instance_method_defined?` returns false.

This started breaking with commit a065bc28d1739631c6177c2995ba345ebcb0c7c8.

#### Steps to reproduce
```ruby
class ParentSerializer < ActiveModel::Serializer
  attributes :$id

  define_method(:$id) do
    object.id
  end
end
```

```ruby
class ChildSerializer < ParentSerializer
  attributes :name
end
```

Throws the following error:
`undefined method '$id' for <Child:0x00...>`

#### Environment

ActiveModelSerializers Version *(commit ref if not on tag)*:  7acbb765bdbf0a6ad8d5bde02aa3648b2b4a48ea

Output of `ruby -e "puts RUBY_DESCRIPTION"`: ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-darwin14.0]

OS Type & Version: OSX 10.11.4

Integrated application and version *(e.g., Rails, Grape, etc)*: Rails

### PR DESCRIPTION
**Pull request by bf4**
Fixes #1653, #1658, #1660

Was introduced in https://github.com/rails-api/active_model_serializers/pull/1650

### CODE REVIEW

File: lib/active_model/serializer.rb [hunk 1]
```diff
@@ -189,7 +176,7 @@ def json_key
     end
 
     def read_attribute_for_serialization(attr)
-      if self.class._serializer_instance_method_defined?(attr)
+      if respond_to?(attr)
```
Comment (NullVoxPopuli) [inline lib/active_model/serializer.rb]: this is waayyy simpler. interesting. Why was it so complicated before?

**Comment by NullVoxPopuli** [comment]
Thanks, I'll give this a go on my repo, and see how it fairs.

**Comment by NullVoxPopuli** [comment]
I can confirm that this works

**Comment by bf4** [comment]
@NullVoxPopuli 

> Thanks, I'll give this a go on my repo, and see how it fairs.

`s/fairs/fares` which I think makes be a sociopath for pointing out per http://www.npr.org/2016/04/02/472716926/panel-round-one

**Comment by NullVoxPopuli** [comment]
ah, yep: http://grammarist.com/usage/fair-fare/ :-)

**Comment by NullVoxPopuli** [comment]
> does pointing other people's errors make you a helpful person or an ass?

I think it's helpful, depending on the tone. Cause it helps reduce my liklihood of being an idiot in the future

**Comment by bf4** [comment]
Added changelog, rebased off of master, and squashed

**Comment by NullVoxPopuli** [comment]
cool, I'll merge once CI passes

**Pull request merged**